plugin/manager.go

```diff
@@ -289,13 +289,54 @@ func attachToLog(id string) func(libcontainerd.IOPipe) error {
 }
 
 func validatePrivileges(requiredPrivileges, privileges types.PluginPrivileges) error {
-	// todo: make a better function that doesn't check order
-	if !reflect.DeepEqual(privileges, requiredPrivileges) {
+	if !isEqual(requiredPrivileges, privileges, isEqualPrivilege) {
 		return errors.New("incorrect privileges")
 	}
+
 	return nil
 }
 
+func isEqual(arrOne, arrOther interface{}, compare func(x, y interface{}) bool) bool {
+	sliceOne := reflect.ValueOf(arrOne)
+	sliceOther := reflect.ValueOf(arrOther)
+	if sliceOne.Kind() != reflect.Slice || sliceOther.Kind() != reflect.Slice {
+		return false
+	}
+
+	if sliceOne.Len() != sliceOther.Len() {
+		return false
+	}
+
+	isContain := false
+	for i := 0; i < sliceOne.Len(); i++ {
+		for j := 0; j < sliceOther.Len(); j++ {
+			if compare(sliceOne.Index(i).Interface(), sliceOther.Index(j).Interface()) {
+				isContain = true
+				break
+			}
+		}
+
+		if !isContain {
+			return false
+		}
+		isContain = false
+	}
+
+	return true
+}
+
+func isEqualPrivilege(one, other interface{}) bool {
+	onePrivilege, ok1 := one.(types.PluginPrivilege)
+	otherPrivilege, ok2 := other.(types.PluginPrivilege)
+	if !ok1 || !ok2 {
```

Comment:
Instead of manually checking types here, why not make this, e.g.

```go
func isEqualPrivilege(a types.PluginPrivilege, b types.PluginPrivilege) bool {
	if a.Name != b.Name.Name {
		return false
	}
	sort.Strings(a.Value)
	sort.Strings(b.Value)

	return reflect.DeepEqual(a.Value, b.Value)
}
```